main.go

```diff
@@ -421,124 +400,139 @@ func mainCore() error {
-	go wiredDBChainMonitor.BlockConnectedHandler()
-	go wiredDBChainMonitor.ReorgHandler()
+		// StakeDatabase
+		wg.Add(2)
+		go sdbChainMonitor.BlockConnectedHandler()
+		go sdbChainMonitor.ReorgHandler()
 
-	if usePG {
 		// dcrsqlite does not handle new blocks except during reorg
 		wg.Add(2)
-		go auxDBChainMonitor.BlockConnectedHandler()
-		go auxDBChainMonitor.ReorgHandler()
-	}
-
-	if cfg.MonitorMempool {
-		mpoolCollector := mempool.NewMempoolDataCollector(dcrdClient, activeChain)
-		if mpoolCollector == nil {
-			return fmt.Errorf("Failed to create mempool data collector")
+		go wiredDBChainMonitor.BlockConnectedHandler()
+		go wiredDBChainMonitor.ReorgHandler()
+
+		if usePG {
+			// dcrsqlite does not handle new blocks except during reorg
+			wg.Add(2)
+			go auxDBChainMonitor.BlockConnectedHandler()
+			go auxDBChainMonitor.ReorgHandler()
 		}
 
-		mpData, err := mpoolCollector.Collect()
-		if err != nil {
-			return fmt.Errorf("Mempool info collection failed while gathering"+
-				" initial data: %v", err.Error())
-		}
+		if cfg.MonitorMempool {
+			mpoolCollector := mempool.NewMempoolDataCollector(dcrdClient, activeChain)
+			if mpoolCollector == nil {
+				return fmt.Errorf("Failed to create mempool data collector")
+			}
 
-		// Store initial MP data
-		if err = baseDB.MPC.StoreMPData(mpData, time.Now()); err != nil {
-			return fmt.Errorf("Failed to store initial mempool data (wiredDB): %v",
-				err.Error())
-		}
+			mpData, err := mpoolCollector.Collect()
+			if err != nil {
+				return fmt.Errorf("Mempool info collection failed while gathering"+
+					" initial data: %v", err.Error())
+			}
+
+			// Store initial MP data
+			if err = baseDB.MPC.StoreMPData(mpData, time.Now()); err != nil {
+				return fmt.Errorf("Failed to store initial mempool data (wiredDB): %v",
+					err.Error())
+			}
+
+			// Setup monitor
+			mpi := &mempool.MempoolInfo{
+				CurrentHeight:               mpData.GetHeight(),
+				NumTicketPurchasesInMempool: mpData.GetNumTickets(),
+				NumTicketsSinceStatsReport:  0,
+				LastCollectTime:             time.Now(),
+			}
 
-		// Setup monitor
-		mpi := &mempool.MempoolInfo{
-			CurrentHeight:               mpData.GetHeight(),
-			NumTicketPurchasesInMempool: mpData.GetNumTickets(),
-			NumTicketsSinceStatsReport:  0,
-			LastCollectTime:             time.Now(),
+			newTicketLimit := int32(cfg.MPTriggerTickets)
+			mini := time.Duration(cfg.MempoolMinInterval) * time.Second
+			maxi := time.Duration(cfg.MempoolMaxInterval) * time.Second
+
+			mpm := mempool.NewMempoolMonitor(mpoolCollector, mempoolSavers,
+				notify.NtfnChans.NewTxChan, quit, &wg, newTicketLimit, mini, maxi, mpi)
+			wg.Add(1)
+			go mpm.TxHandler(dcrdClient)
 		}
+		return nil
+	}
 
-		newTicketLimit := int32(cfg.MPTriggerTickets)
-		mini := time.Duration(cfg.MempoolMinInterval) * time.Second
-		maxi := time.Duration(cfg.MempoolMaxInterval) * time.Second
+	// do not collect the data summary if the sync status page has been activated.
```

Comment:
`collectBlocksAfterSyncing` does two main things:
1. configures and starts handlers such as new blocks, reorg, mempool.
2. Initial data collection for the explorer.

The naming of `collectBlocksAfterSyncing` should change to reflect both the start of handlers, and the initial data collection.  Also, the `AfterSyncing` is not appropriate for the function name as it does not know about sync, it just performs these two functions.

// do not collect the data summary if the sync status page has been activated.